tests/unit/s2n_self_talk_nonblocking_test.c

```diff
@@ -212,10 +316,30 @@ int main(int argc, char **argv)
     EXPECT_EQUAL(status, 0);
     EXPECT_SUCCESS(s2n_config_free(config));
     EXPECT_SUCCESS(s2n_cert_chain_and_key_free(chain_and_key));
+
+    if (iov) {
+        for (int i = 0; i < iov_size; i++) {
+            free(iov[i].iov_base);
+        }
+        free(iov);
+    }
+
+    return 0;
+}
+
+int main(int argc, char **argv)
+{
+#define test_count _test_count
+    BEGIN_TEST();
+#undef test_count
```

Comment:
That looks like a hack for the thing which might be useful in other tests (: How about moving `test_count` into static variable in test header file instead?

Reply:
I don't think having a static variable on a header file is good idea. What we could do is to have it declared in some .c file and have an extern declaration of it on the header. That being said, I don't think other tests can benefit from this though. There aren't other APIs that have this type of variations where you can put 2 similar looking tests into a single file.

Reply:
> I don't think having a static variable on a header file is good idea. What we could do is to have it declared in some .c file and have an extern declaration of it on the header.

I don't expect tests having more then one compilation unit, so static variable should be fine. extern is better, but requires 1 more .c file to be linked with all the tests, so thought it's simple to keep it static for now.

> I don't think other tests can benefit from this though. There aren't other APIs that have this type of variations where you can put 2 similar looking tests into a single file.

I think having tests testing 2 similar APIs is only a part of the problem. I can totally see us hitting the same problem by trying to move some of the `EXPECT_*` into a function to avoid repeating the same pattern in unit tests. That said, I'm ok leaving this improvement to the next PR needing this functionality.